src/test/java/com/buinak/imageapi/controller/ImageControllerIT.java

```diff
@@ -30,38 +34,67 @@
     @Autowired
     ImageRepository imageRepository;
 
-    final static String IMG_PATH = "src/test/java/com/buinak/imageapi/controller/testimg.jpeg";
+    @Autowired
+    ImageService imageService;
 
 
     @Test
-    public void addImage() throws IOException {
-        MultipartFile multipartFile = new MockMultipartFile("testimg.jpeg", new FileInputStream(new File(IMG_PATH)));
+    public void addImage() {
+        MockMultipartFile multipartFile = new MockMultipartFile("data", "testimg.jpeg", "text/plain", "some xml".getBytes());
 
         imageController.addImage("NAME1", "DESC1", multipartFile);
 
-        ImageRepository.ImageInformationView imageInformationView = imageController.findImageByName("NAME1").getBody();
-        assertThat(imageInformationView.getName()).isEqualTo("NAME1");
-        assertThat(imageInformationView.getDescription()).isEqualTo("DESC1");
+        Image image= imageController.findImageByName("NAME1").getBody();
+        assertThat(image.getName()).isEqualTo("NAME1");
+        assertThat(image.getDescription()).isEqualTo("DESC1");
+
+        imageService.deleteImage(image.getId());
```

Comment:
this is better done with a @Before or @After annotation, if your tests have to clean up the DB. Why clean it up though? I don't see the point

Reply:
I want all the tests to work individually. If I don't delete the images after then the list link test will have a different number of links generated. But I can simply get rid of that test and it will work.

Reply:
I understand, but it's a bad practice. It gets very bad when you introduce concurrent execution of tests -- since tests work with shared context -- for instance

Reply:
However, @After annotation is created specifically to solve the problem you're solving, but in a very fool-proof and centralised manner :)